### PR TITLE
All: Use mypy for every projects

### PR DESCRIPTION
**Pull request by GuillaumeLescure**

### CODE REVIEW

**Comment by GuillaumeLescure** [comment]
Expalantions about the exclude paths :
* /monitor/ is dead code
* /third_party/ we cannot modify third party code so it's not relevant to check it
* /tests/ not relevant to check tests (questionable but for now it's ok)
* /sql/ not relevant to check sql (questionable but for now it's ok)
* env.py and setup.py should be checked but mypy return a false-positif error (duplicate module)

**Comment by GuillaumeLescure** [comment]
More informations about mypy :

* type checking only works when the inline comment is done, ex : # type: () -> None
(for now, it's not done in all codes yet)

* type check only works if the function has the type check comment, ex :
```
def run(): # type: () -> None
   x = "ok" # type: typeUnknown
   print "jormun", type(x)
   greet_world(x)
```
✘ $ error: Name 'typeUnknown' is not defined (edited)

```
def run():
  x = "ok" # type: typeUnknown
  print "jormun", type(x)
  greet_world(x)
```
✔ $ all good !

* some files seems to be ignored if the file "\_\_init\_\_.py" is present or missing, ex :
A type error is introduced in Jormun but the function is defined in Eitri:

| \_\_init.py\_\_ in Jormun | \_\_init.py\_\_ in Eitri | Result |
| --- | --- | --- |
| Present | Present | ✔ : no error found |
| Present | Missing | ✘ : error found |
| Missing | Present | ✔ : no error found |
| Missing | Missing | ✘ : error found |

The archive with the tests : https://github.com/CanalTP/navitia/files/3320974/test_mypy.zip

File: source/eitri/ed_handler.py [hunk 1]
```diff
@@ -30,7 +30,6 @@
 import glob
 import os
 from navitiacommon import utils, launch_exec
-from navitiacommon.launch_exec import launch_exec
```
Comment (kinnou02) [inline source/eitri/ed_handler.py:33]: Are you sure that eitri still works after that?

**Review by kinnou02 — COMMENTED** [review]
`monitor` isn't dead code, it's deployed on production and used by fabric and the monitoring platform

File: source/eitri/ed_handler.py (hunk 1 above)
Comment (GuillaumeLescure) [reply source/eitri/ed_handler.py:33]: You're right I have this error :
'module' object is not callable

I'm investigating ...

Comment (GuillaumeLescure) [reply source/eitri/ed_handler.py:33]: Fix done.

**Comment by kinnou02** [comment]
Jenkins, test this please.

**Comment by GuillaumeLescure** [comment]
retest this please

**Review by kinnou02 — APPROVED** [review]

File: source/eitri/ed_handler.py [hunk 2]
```diff
@@ -30,7 +30,6 @@
 import glob
 import os
 from navitiacommon import utils, launch_exec
```
Comment (mbounabi) [inline source/eitri/ed_handler.py:32]: from utils, only 1 function used, from launch_exec, only one function used. Not sure about performance (I'll try to do some tests or find doc about it), but I'm not a fan of 'launch_exec.launch_exec'
```suggestion
from navitiacommon.utils import type_of_data 
from navitiacommon.launch_exec import launch_exec
```

**Review by mbounabi — APPROVED** [review]

**Review by nonifier — APPROVED** [review]

**Pull request merged**